core/src/main/scala/kafka/server/DynamicConfig.scala

```diff
@@ -141,6 +144,18 @@ object DynamicConfig {
     def names = ipConfigs.names
 
     def validate(props: Properties) = DynamicConfig.validate(ipConfigs, props, customPropsAllowed = false)
+
+    def validateIpOrHost(ip: String): Unit = {
+      if (ip != ConfigEntityName.Default) {
+        if (!Utils.validHostPattern(ip))
+          throw new InvalidRequestException(s"$ip is not a valid hostname")
```

Comment:
I wonder if this step is really necessary as `InetAddress.getByName` also ensures that the hostname is valid. What do you think?